src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py

```diff
@@ -869,6 +869,91 @@ def test_group_deployment_thru_uri(self, resource_group):
                  checks=self.is_empty())
 
 
+class DeploymentScriptsTest(ScenarioTest):
+    @ResourceGroupPreparer(name_prefix='cli_test_deployment_scripts')
+    def test_list_all_deployment_scripts(self, resource_group):
+        curr_dir = os.path.dirname(os.path.realpath(__file__))
+        self.kwargs.update({
+            'deployment_script_name': self.create_random_name('script', 20),
+            'deployment_name': self.create_random_name('ds', 20),
+            'resource_group': resource_group,
+            'template_file': os.path.join(curr_dir, 'deployment-scripts-deploy.json').replace('\\', '\\\\'),
+        })
+
+        count = self.cmd("deployment-scripts list --query \"length([?name=='{deployment_script_name}'])\"").get_output_in_json() or 0
```

Comment:
Could you please also assertion for count with JMESPathCheck?

Reply:
Code updated to assert count = 0